### PR TITLE
Initial Tron Guy Labs keyboard implementation.

### PR DESCRIPTION
**Pull request by jmaynard**
## Description

This PR adds support for the Tron Guy Labs replacement controller for the IBM M122 terminal keyboard. The layout is intended to provide as much 3270-like functionality as possible, to match the 3270 keyboard legends. There are two variants; one for the  Teensy ++ 2.0 MCU, and one for the STM32F411 BlackPill. The two should function the same at the user level.
The changes are confined to the keyboards/tronguylabs directory.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [X] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by zvecr — CHANGES_REQUESTED** [review]
All hardware files (*.pdf, *.zip, LICENSE) need to be removed.

You can host them in another repo, and link to them in the boards readme if you wish.

File: keyboards/tronguylabs/tronguylabs.c [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "tronguylabs.h"
+
+bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
+	// This is here because we actually do some things on various key actions
+	return process_record_user(keycode, record);
+}
```
Comment (zvecr) [inline keyboards/tronguylabs/tronguylabs.c:21]:
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]
I would also advise restructuring.

`keyboards/tronguylabs/` is your vendor folder and should not contain keyboard specific files like `tronguylabs.h` `info.json` `rules.mk` and should be moved to `keyboards/tronguylabs/m122_3270`

the rest would become something like....
```
keyboards/tronguylabs/m122_3270/teensy/
keyboards/tronguylabs/m122_3270/blackpill/
keyboards/tronguylabs/m122_3270/keymaps/default/
```

File: keyboards/tronguylabs/tronguylabs.c (hunk 1 above)
Comment (jmaynard) [reply keyboards/tronguylabs/tronguylabs.c]: Removed as suggested.

File: keyboards/tronguylabs/m122_3270/config.h [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "config_common.h"
+
+#define VENDOR_ID       0x1209
```
Comment (drashna) [inline keyboards/tronguylabs/m122_3270/config.h:21]: pid.codes vendor id 👍🏻

File: keyboards/tronguylabs/m122_3270/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+enum custom_keycodes {
+    PLACEHOLDER = SAFE_RANGE,
+
+    JM_DOT,       // . both shifted and unshifted
```
Comment (drashna) [inline keyboards/tronguylabs/m122_3270/keymaps/default/keymap.c:21]:
```suggestion
    JM_DOT = SAFE_RANGE, // . both shifted and unshifted
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/tronguylabs/m122_3270/config.h (hunk 2 above)
Comment (jmaynard) [reply keyboards/tronguylabs/m122_3270/config.h:21]: It's on request there, but they haven't merged the PR yet...if they turn it down, I'll pick something else.

File: keyboards/tronguylabs/info.json [hunk 4]
```diff
@@ -0,0 +1,134 @@
+{
```
Comment (zvecr) [inline keyboards/tronguylabs/info.json:1]: Still needs to be moved to `keyboards/tronguylabs/m122_3270/info.json`

Comment (jmaynard) [reply keyboards/tronguylabs/info.json:1]: The readme, too?

Comment (zvecr) [reply keyboards/tronguylabs/info.json]: yeah, missed that one off somehow.

**Review by zvecr — CHANGES_REQUESTED** [review]
Otherwise, `readme.md` would need updating with
```
Make example for this keyboard (after setting up your build environment):

    make tronguylabs/m122-3270/blackpill:default
    make tronguylabs/m122-3270/teensy:default
```

**Review by zvecr — APPROVED** [review]

File: keyboards/tronguylabs/m122_3270/config.h (hunk 2 above)
Comment (zvecr) [reply keyboards/tronguylabs/m122_3270/config.h:21]: Ive added the `awaiting PR` label while this is sill unresolved.

File: keyboards/tronguylabs/m122_3270/rules.mk [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # no audio output
```
Comment (fauxpark) [inline keyboards/tronguylabs/m122_3270/rules.mk:16]:
```suggestion
AUDIO_ENABLE = no           # Audio output
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

File: keyboards/tronguylabs/m122_3270/config.h (hunk 2 above)
Comment (drashna) [reply keyboards/tronguylabs/m122_3270/config.h:21]: For quick reference: https://github.com/pidcodes/pidcodes.github.com/pull/656

**Comment by drashna** [comment]
Thanks!

**Pull request merged**